TCAT/Controllers/HomeMapViewController.swift

```diff
@@ -0,0 +1,251 @@
+            make.height.equalTo(optionsCardVC.calculateCardHeight())
+        }
+    }
+    
+    /// Show a temporary loading screen
+    func showLoadingScreen() {
+        
+        loadingView.backgroundColor = Colors.backgroundWash
+        view.addSubview(loadingView)
+        
+        loadingView.snp.makeConstraints { (make) in
+            make.top.bottom.leading.trailing.equalToSuperview()
+        }
+        
+        let indicator = LoadingIndicator()
+        loadingView.addSubview(indicator)
+        indicator.snp.makeConstraints { (make) in
+            make.center.equalToSuperview()
+            make.size.equalTo(loadingIndicatorSize)
+        }
+        
+    }
+    
+    func removeLoadingScreen() {
+        loadingView.removeFromSuperview()
+        viewWillAppear(false)
+    }
+}
+
+// MARK: Location Delegate
+extension HomeMapViewController: CLLocationManagerDelegate {
+    
+    func locationManager(_ manager: CLLocationManager, didChangeAuthorization status: CLAuthorizationStatus) {
+        
+        if status == .denied {
+            let alertTitle = Constants.Alerts.LocationDisabled.title
+            let alertMessage = Constants.Alerts.LocationDisabled.message
+            let alertController = UIAlertController(title: alertTitle, message: alertMessage, preferredStyle: .alert)
+            let settingsAction = UIAlertAction(title: Constants.Alerts.LocationDisabled.settings, style: .default) { (_) in
+                UIApplication.shared.open(URL(string: UIApplication.openSettingsURLString)!, options: convertToUIApplicationOpenExternalURLOptionsKeyDictionary([:]), completionHandler: nil)
+            }
+            
+            guard let showReminder = userDefaults.value(forKey: Constants.UserDefaults.showLocationAuthReminder) as? Bool else {
+                userDefaults.set(true, forKey: Constants.UserDefaults.showLocationAuthReminder)
+                let cancelAction = UIAlertAction(title: Constants.Alerts.LocationDisabled.cancel, style: .default, handler: nil)
+                alertController.addAction(cancelAction)
+                alertController.addAction(settingsAction)
+                alertController.preferredAction = settingsAction
+                present(alertController, animated: true)
+                return
+            }
+            
+            if !showReminder {
+                return
+            }
+            
+            let dontRemindAgainAction = UIAlertAction(title: Constants.Alerts.LocationDisabled.cancel, style: .default) { _ in
+                self.userDefaults.set(false, forKey: Constants.UserDefaults.showLocationAuthReminder)
+            }
+            alertController.addAction(dontRemindAgainAction)
+            
+            alertController.addAction(settingsAction)
+            alertController.preferredAction = settingsAction
+            
+            present(alertController, animated: true)
```

Comment:
lines 192 - 214 are a bit hard to read. might be clearer if you did something like
```
if let showReminder = userDefaults.value(forKey: Constants.UserDefaults.showLocationAuthReminder) as? Bool {
    if showReminder {
        ...
    }
    return
}

userDefaults.set(true, forKey: Constants.UserDefaults.showLocationAuthReminder)
let cancelAction = UIAlertAction(title: Constants.Alerts.LocationDisabled.cancel, style: .default, handler: nil)
alertController.addAction(cancelAction)
alertController.addAction(settingsAction)
alertController.preferredAction = settingsAction
present(alertController, animated: true)
```